File: dataset/human/python/file_64.py
import sys
def input(): return sys.stdin.readline().strip()

mi = lambda: map(int, input().split())

R, G, B, N = mi()

count = 0
for i in range(N//R + 1):
    for j in range((N-i*R)//G + 1):
        if (N - i*R - j*G) % B == 0:
            count += 1

print(count)

File: dataset/human/c/file_64.c
#include<stdio.h>

int main(void){
  int R, G, B, N;
  scanf("%d %d %d %d", &R , &G , &B , &N);

  int count = 0;
  for(int r = 0 ; r <= 3000 ; r++){
    for(int g = 0 ; g <= 3000 ; g++){

      int v = r*R + g*G;
      if(N >= v && (N - v) % B == 0)   count++;

    }
  }

  printf("%d", count);

return 0;
}
